addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CookieHttpOnlyScanner.java

```diff
@@ -73,6 +73,9 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
         while (iterator.hasNext()) {
             String headerValue = (String) iterator.next();
             if (!CookieUtils.hasAttribute(headerValue, HTTP_ONLY_COOKIE_ATTRIBUTE)) {
+                if (CookieUtils.isExpired(headerValue)) {
+                    return;
```

Comment:
Should continue, we should still check the other cookies (worth adding a test for this case).